File: tests/fixtures/sample.ts
interface Greetable {
  greet(name: string): string;
}

class Person implements Greetable {
  private name: string;

  constructor(name: string) {
    this.name = name;
  }

  greet(name: string): string {
    return `Hello, ${name}, I'm ${this.name}`;
  }
}

function makeGreeter(name: string): Person {
  return new Person(name);
}

function main(): void {
  const p = makeGreeter("Alice");
  const msg = p.greet("Bob");
  console.log(msg);
}

const DEFAULT_NAME: string = "World";
type NameProvider = () => string;

File: tests/fixtures/sample.rs
use std::fmt;

trait Greetable {
    fn greet(&self) -> String;
}

struct Person {
    name: String,
}

impl Greetable for Person {
    fn greet(&self) -> String {
        format!("Hello, {}!", self.name)
    }
}

fn new_person(name: &str) -> Person {
    Person { name: name.to_string() }
}

fn main() {
    let p = new_person("World");
    let msg = p.greet();
    println!("{}", msg);
}

const DEFAULT_NAME: &str = "World";
